src/startup.ts

```diff
@@ -1,60 +1,106 @@
-import { getClient } from "./clients/TorrentClient.js";
-import { MatchMode } from "./constants.js";
 import { CrossSeedError } from "./errors.js";
-import { logger } from "./logger.js";
-import { getRuntimeConfig } from "./runtimeConfig.js";
-import { validateTorrentDir } from "./torrent.js";
+import { Label, logger } from "./logger.js";
+import { Action } from "./constants.js";
 import { validateTorznabUrls } from "./torznab.js";
-import { validateAction } from "./action.js";
+import { getClient } from "./clients/TorrentClient.js";
+import {
+	RuntimeConfig,
+	getRuntimeConfig,
+	setRuntimeConfig,
+} from "./runtimeConfig.js";
+import { inspect } from "util";
+import { stat, access } from "fs/promises";
+import { constants } from "fs";
+
+/**
+ * validates existence, permission, and that a path is a directory
+ * @param path string of path to validate
+ * @param optionName name of the configuration key
+ * @returns true if path exists and has required permission
+ */
+async function verifyPath(
+	path: string,
+	optionName: string,
+	permissions: number
+): Promise<boolean> {
+	try {
+		if ((await stat(path)).isDirectory()) {
+			await access(path, permissions);
+			return true;
+		}
+	} catch (error) {
+		if (error.code === "ENOENT") {
+			logger.error(
+				`\tYour ${optionName} "${path}" is not a valid directory on the filesystem.\n`
+			);
+		} else {
+			logger.error(
+				`\tYour ${optionName} "${path}" has invalid permissions.`
+			);
+		}
+	}
+	return false;
+}
+
+/**
+ * verifies the config paths provided against the filesystem
+ * @returns true (if paths are valid)
+ */
+async function checkConfigPaths(): Promise<void> {
+	const { action, linkDir, dataDirs, torrentDir, outputDir } =
+		getRuntimeConfig();
+	let pathFailure: number = 0;
+
+	if (!(await verifyPath(torrentDir, "torrentDir", constants.R_OK))) {
+		pathFailure++;
+	}
 
-function validateOptions() {
-	const {
-		action,
-		rtorrentRpcUrl,
-		qbittorrentUrl,
-		transmissionRpcUrl,
-		delugeRpcUrl,
-		dataDirs,
-		linkDir,
-		matchMode,
-		skipRecheck,
-	} = getRuntimeConfig();
 	if (
-		action === "inject" &&
-		!(
-			rtorrentRpcUrl ||
-			qbittorrentUrl ||
-			transmissionRpcUrl ||
-			delugeRpcUrl
-		)
+		action == Action.SAVE &&
```

Comment:
rTorrent injection needs outputDir access as well

```suggestion
		(action == Action.SAVE || rtorrentRpcUrl) &&
```

Reply:
Should we handle outputDir not being set in Zod's check for injection settings at the bottom of the schema? It currently only checks if <client>Url's are set with injection....

I can add a custom message or just include in the error `(and outputDir)` after the rtorrentRpcUrl

Reply:
I guess we require outputDir so that seems a little pointless actually.